test/server/drain_manager_impl_test.cc

```diff
@@ -98,24 +128,27 @@ TEST_F(DrainManagerImplTest, DrainDeadlineProbability) {
   EXPECT_FALSE(drain_manager.drainClose());
   drain_manager.startDrainSequence([] {});
 
-  // random() should be called when elapsed time < drain timeout
-  EXPECT_CALL(server_.random_, random()).Times(2);
-  // Current elapsed time is 0
-  // drainClose() will return true when elapsed time > (4 % 3 == 1).
-  EXPECT_FALSE(drain_manager.drainClose());
-  simTime().advanceTimeWait(std::chrono::seconds(2));
-  EXPECT_TRUE(drain_manager.drainClose());
-  simTime().advanceTimeWait(std::chrono::seconds(1));
-  EXPECT_TRUE(drain_manager.drainClose());
+  if (drain_gradually) {
+    // random() should be called when elapsed time < drain timeout
+    EXPECT_CALL(server_.random_, random()).Times(2);
+    // Current elapsed time is 0
+    // drainClose() will return true when elapsed time > (4 % 3 == 1).
+    EXPECT_FALSE(drain_manager.drainClose());
+    simTime().advanceTimeWait(std::chrono::seconds(2));
+    EXPECT_TRUE(drain_manager.drainClose());
+    simTime().advanceTimeWait(std::chrono::seconds(1));
+    EXPECT_TRUE(drain_manager.drainClose());
+  } else {
+    EXPECT_CALL(server_.random_, random()).Times(0);
+    EXPECT_TRUE(drain_manager.drainClose());
+    simTime().advanceTimeWait(std::chrono::seconds(2));
+    EXPECT_TRUE(drain_manager.drainClose());
+    simTime().advanceTimeWait(std::chrono::seconds(1));
+    EXPECT_TRUE(drain_manager.drainClose());
+  }
 }
 
-TEST_F(DrainManagerImplTest, ModifyOnly) {
-  InSequence s;
-  DrainManagerImpl drain_manager(server_, envoy::config::listener::v3::Listener::MODIFY_ONLY);
-
-  EXPECT_CALL(server_, healthCheckFailed()).Times(0); // Listener check will short-circuit
-  EXPECT_FALSE(drain_manager.drainClose());
-}
+INSTANTIATE_TEST_SUITE_P(DrainGradually, DrainManagerImplTest, testing::Bool());
```

Comment:
Is DrainGradually the correct name for this?  Perhaps DrainStrategies?

Reply:
Done.